Packages/MIES/MIES_SweepFormula.ipf

```diff
@@ -12,41 +12,117 @@ Function FormulaParser(formula)
 	String formula
 
 	Variable i
-	String token
-	String jsonPath = ""
+	String lastToken = "", token
+	Variable argument, parameters
+	String jsonPath = "", str = "", num = ""
+	String state = "", lastState = "", basePath
 	Variable jsonID = JSON_New()
+	Variable childID, dummy
 
 	if(strlen(formula) == 0)
 		return jsonID
 	endif
 
-	for(i = 0; i < strlen(formula) - 1; i += 1)
+	for(i = 0; i < strlen(formula); i += 1)
 		token = formula[i]
+
+		// collect
+		if(GrepString(token, "[A-Za-z0-9_]"))
+			str += token
+			continue
+		endif
+
+		// state
 		strswitch(token)
+			case "(":
+				if(strlen(str) > 0)
+					state = "function"
+				else
+					state = "group"
+				endif
+				break
 			case "/":
-				token = EscapeJsonPath(token)
 			case "*":
-				if(strsearch(formula, "+", 0) > -1)
+				if((strsearch(formula, "+", i) != -1) || (strsearch(formula, "-", i) != -1))
+					str += token // look ahead for conflicts
 					continue
 				endif
-				if(strsearch(formula, "-", 0) > -1)
+
+				if(!cmpstr(state, "+") || !cmpstr(state, "-"))
+					str += token // preserve state
 					continue
 				endif
 			case "+":
 			case "-":
-				if(!!cmpstr(jsonPath, token))
-					jsonPath = token
-					JSON_AddTreeArray(jsonID, jsonPath)
+				if(!cmpstr(state, "function") || !cmpstr(state, "argument"))
+					str += token
+					continue
 				endif
-				JSON_AddJSON(jsonID, jsonPath, FormulaParser(formula[-inf, i - 1]))
-				JSON_AddJSON(jsonID, jsonPath, FormulaParser(formula[i + 1, inf]))
-				return jsonID
+				state = token
+				break
+			case ")":
+				state = "endfunction"
+				argument = 0
+				break
+			case ",":
+				state = "argument"
+				break
 		endswitch
+
+		// path action
+		if(!!cmpstr(state, lastState))
+			if(!cmpstr(state, "+") && !cmpstr(lastState, "*") && !cmpstr(jsonPath, "/" + lastState))
+				print "not working"
+			endif
+
+			if(JSON_GetType(jsonID, jsonPath) == JSON_ARRAY)
+				JSON_AddObjects(jsonID, jsonPath)
```

Comment:
This is the usual `JSON_AddTreeArray` issue where the automatic creation of objects is not allowed.